### PR TITLE
refactor(es/transforms/compat): visit_mut for static_blocks

### PR DESCRIPTION
**Pull request by kwonoj**
Continuing #2370.

:Ez: one

I thought I picked file to continue https://github.com/swc-project/swc/pull/2966 but realized it's not after complete it. Anyway, this is small module so I took a stab before going back to #2966. PR applies few refactorings as visit_mut didn't need to follow strictly same path what fold does for this case.

### CODE REVIEW

File: crates/swc_ecma_transforms_compat/src/es2022/static_blocks.rs [hunk 1]
```diff
@@ -1,50 +1,47 @@
...
-        Class {
-            body: members,
-            ..class
         }
     }
-    fn fold_static_block(&mut self, static_block: StaticBlock, private_id: JsWord) -> PrivateProp {
+
+    fn visit_mut_static_block(
+        &mut self,
+        static_block: &mut StaticBlock,
```
Comment (kdy1) [inline crates/swc_ecma_transforms_compat/src/es2022/static_blocks.rs:42]: I think it's better to take `StaticBlock` without reference, as we always `.take()` it.

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you!

**Review by kdy1 — APPROVED** [review]
LGTM. Thanks!

**Comment by sosukesuzuki** [comment]
I've implemented static_blocks transformation and tried this refactoring in the past and gave up, thanks!

**Comment by kwonoj** [comment]
@sosukesuzuki 👍

**Pull request merged**